app/app.js

```diff
@@ -3,177 +3,49 @@
-        if (fileType.includes('png')) {
-            return false // image is a PNG
-        } else if (fileType.includes('jpeg')) {
-            return true // image is a JPEG
-        } else {
-            let error = new Error('File type isn\'t PNG or JPEG. Only JPEG and PNG images are allowed.')
-            error.code = 400
-            return error // image is not PNG or JPEG, so it cannot be used
-        }
-    }
-}
-
-/**
- * Handles passing the base64 string to all proper methods, which includes checking its file type and writing the
- * file to disk
- * @param {string} base64String - base64 string of file
- * @returns {*} - file path if the argument provided was PNG or JPEG, null otherwise
- */
-function processImage(base64String) {
-    if (base64String !== null && base64String !== undefined && base64String !== '') {
-        const fileType = checkIfBase64IsJpeg(base64String) // checking the file type of the base64
-        // throws an error if file type is not compatible
-        if (fileType instanceof Error) {
-            return fileType
-        } else {
-            const fileName = uuid() + '.jpeg'
-            // call writeJpegToDisk if true or writePngToDisk if false
-            let writeFile = fileType ? writeJpegToDisk(base64String, fileName) : writePngToDisk(base64String, fileName)
-            if (writeFile instanceof Error) {
-                throw writeFile
-            } else {
-                return '../slothbucket/' + fileName // file path
-            }
-        }
-    } else {
-        let error = new Error('Cannot process null, undefined, or empty string arguments')
-        error.code = 400
-        return error
-    }
-}
-
-// HTTP POST for any TensorFlow image recognition with bonus sloth check
-app.post('/sloth-check', (req, res) => {
-    if (req.body.base64 !== null && req.body.base64 !== undefined && req.body.base64 !== '') {
-        let filePath = processImage(req.body.base64) // sending to check file type and save to disk
-        if (filePath instanceof Error) {
-            // if there's an error, that means the image wasn't able to be processed (e.g., it wasn't a JPEG or PNG)
-            res.status(200).send({status: filePath.code, message: filePath.message})
-        } else {
-            // commands to run the TensorFlow script
-            child_process.execFile('python', ['classify_image.py', '--image_file', filePath], {
-                cwd: '../root' // change working directory
-            }, function(error, result) {
-                if (error !== null) {
-                    res.status(500)
-                        .send({status: 500, message: 'Internal error encountered while running TensorFlow script.'})
+// routes ===============================
+app.post('/sloth-check', (request, response) => {
+    const base64 = request.body.base64
+    if (base64) {
+        // first, write the file to disk
+        writeFile.writeFileToDisk(base64).then((fileName) => {
```

Comment:
In Node, it's typical for callback functions to be in the form `function(error, result)`. This gives consumers the ability to deal with errors. Since you probably want to bubble up errors to the route handling logic, you'll want to be able to receive errors from any function calls here

Reply:
Disregard, didn't notice this was a promise